packages/react/react/src/hooks/useInterval.ts

```diff
@@ -29,10 +37,10 @@ export function useInterval(callback: () => void, options: IntervalOptions) {
     }
 
     function tick() {
-      savedCallback();
+      if (enabled) savedCallback();
     }
 
     const id = window.setInterval(tick, delay);
     return () => window.clearInterval(id);
-  }, [delay, savedCallback]);
+  }, [delay, savedCallback, enabled]);
```

Comment:
I think it would be better for us if we did not register a callback at all if it is not enabled like the following code. 

```
useEffect(() => {
  if (delay == null || !enabled) {
    return;
  }

  const id = window.setInterval(tick, delay);
  return () => window.clearInterval(id);
}, [delay, savedCallback, enabled]);
```

Reply:
Thank you for your feedback. I have reflected your opinion and added the corresponding test code.